src/service/firestoreApi.ts

```diff
@@ -19,22 +21,41 @@ const subscribeToPlaylist = async (id: string, observer: (playlist: any) => void
 
 const getUserOwnPlayLists = async (userId: string) => {
     return database
-    .collection('playlists')
-    .where('owner', '==', userId)
-    .get()
-    .then((querySnapshot) => {
-        let data: any = []
-        querySnapshot.forEach((doc) => {
-            data = [...data, { id: doc.id, ...doc.data() }]
-        });
-        return data
+        .collection('playlists')
+        .where('owner', '==', userId)
+        .get()
+        .then((querySnapshot) => {
+            let data: any = []
+            querySnapshot.forEach((doc) => {
+                data = [...data, { id: doc.id, ...doc.data() }]
+            });
+            return data
 
-    })
+        })
+}
+
+const subscribeToSongsCollection = async (id: string, observer: (playlist: any) => void) => {
+    const callback = (snapshot: any) => {
+        console.log(snapshot.docs)
+        const songList = snapshot.docs.map((doc: any) => {
+            const song = { ...doc.data(), id: doc.id }
+            return song
+        })
+        console.log(songList)
+        observer(songList)
+    }
+    await database.collection('playlists').doc(id).collection('songs').onSnapshot(callback)
+}
```

Comment:
can you seperate the chained funcitons with a new line?
.foo()
.bar()
.baz()

Reply:
fixed